flexmeasures/data/services/sensors.py

```diff
@@ -197,3 +198,77 @@ def build_sensor_status_data(
             sensor_status["asset_name"] = asset.name
             sensors.append(sensor_status)
     return sensors
+
+
+def build_asset_jobs_data(
+    asset: Asset,
+) -> list[dict]:
+    """Get all jobs data for an asset
+    Returns a list of dictionaries, each containing the following keys:
+    - job_id: id of a job
+    - job_type: type of a job (scheduling or forecasting)
+    - sensor_or_asset_class: type of an asset that is linked to the job (asset or sensor)
+    - asset_id: id of sensor or asset
+    - status: job status (e.g finished, failed, etc)
+    - err: job error (equals to None when there was no error for a job)
+    """
+
+    jobs = list()
+
+    # try to get scheduling jobs for asset first (only scheduling jobs can be stored by asset id)
+    jobs.append(
+        (
+            "scheduling",
+            "asset",
+            asset.id,
+            current_app.job_cache.get(asset.id, "scheduling", "asset"),
+        )
+    )
+
+    for sensor in asset.sensors:
+        jobs.append(
+            (
+                "scheduling",
+                "sensor",
+                sensor.id,
+                current_app.job_cache.get(sensor.id, "scheduling", "sensor"),
+            )
+        )
+        jobs.append(
+            (
+                "forecasting",
+                "sensor",
+                sensor.id,
+                current_app.job_cache.get(sensor.id, "forecasting", "sensor"),
+            )
+        )
+
+    jobs_data = list()
+    for job_type, sensor_or_asset_class, asset_id, jobs in jobs:
```

Comment:
Let's add a comment above this loop here, to make clear what it's for:

"Building the actual return list - we also unpack lists of jobs, each to its own entry, and we add error info"